transport/src/main/java/io/zeebe/transport/impl/sender/Sender.java

```diff
@@ -198,8 +188,13 @@ private void onRequestSubmitted(final OutgoingRequest request) {
           actor.runDelayed(Duration.ofMillis(10), () -> submittedRequests.offer(request));
         }
       } else {
-        // no remote address available, retry
-        actor.runDelayed(Duration.ofMillis(10), () -> submittedRequests.offer(request));
+        if (request.shouldRetry(
```

Comment:
I can only see two usages for the predicate, here when we have no remote address, and when we try to complete the request and an error has occurred. 

This makes me think maybe for the 0.24 implementation we can avoid conflating the two and have the "should-retry-if-no-remote-handler-found" more as a property of the request than part of the predicate itself. Anyway we can discuss that later.